src/base/projects/PeerSelector.svelte

```diff
@@ -1,14 +1,29 @@
 <script lang="ts">
-  import { createEventDispatcher } from "svelte";
+  import { createEventDispatcher, onMount } from "svelte";
   import Icon from "@app/Icon.svelte";
   import Dropdown from "@app/Dropdown.svelte";
   import { formatSeedId } from "@app/utils";
+  import type { Peer } from "@app/project";
 
   export let peer: string | null = null;
-  export let peers: (string | { id: string })[];
+  export let peers: Peer[];
   export let toggleDropdown: (input: string) => void;
   export let peersDropdown = false;
 
+  let meta: Peer | undefined;
+  // List of items to be created for the Dropdown component.
+  let items: { key: string; value: string; badge: string | null }[] = [];
+
+  onMount(() => {
+    meta = peers.find(p => p.id === peer);
+    items = peers.map(p => {
+      if (! p.name) console.debug("Not able to resolve peer identity for: ", p.id);
+      let key = p.name ? `${p.name} ${p.id}` : p.id;
+
+      return { key, value: p.id, badge: p.delegate ? "delegate" : null };
+    });
+  });
```

Comment:
Is there a reason these vars are set in `onMount` vs. just being set directly as consts?

Reply:
Hmm I wanted to be explicit when the setting should happen, but it could be also outside of a `onMount` callback

Reply:
Ok, we can change it in a later commit. Looks good!